### PR TITLE
Add benchmarks

### LINKED ISSUE #180: Do benchmark operations of the document
**Description**:

Do benchmark operations of the document.

https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go

**Why**:

We can find the bottlenecks of operations and perform performance optimization.

### PR DESCRIPTION
**Pull request by zeikar**
**What this PR does / why we need it**:
Benchmark operations of the document.

**Which issue(s) this PR fixes**:

Fixes #180 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation**:

```docs
editing-trace.json is based on https://github.com/automerge/automerge-perf.
```

**Checklist**:

### CODE REVIEW

**Comment by zeikar** [comment]
If you run BenchmarkEditing (about 200,000 data), it won't finish even if you run it for several hours.
So, it is necessary to check whether the benchmark is properly implemented.
If the benchmark code is ok, it seems that Yorkie has performance issues.

**Comment by dc7303** [comment]
When I run the `make bench` command, an error occurs.  This is happening when I add the `-bench` option on running the go test. 😨
Are any other reviewers okay? First, I'll fix this issue and do a review...

```bash
go test -tags bench -bench=./test/bench
no Go files in /Users/scott/Workspace/dc7303/yorkie
make: *** [bench] Error 1
```

**Comment by hackerwins** [comment]
@dc7303

We can run benchs like below.

```bash
cd test/bench
go test -tags=bench -bench=.
```

File: Makefile [hunk 1]
```diff
@@ -38,6 +38,9 @@ test:
 	go clean -testcache
 	go test -tags integration -race ./...
 
+bench:
+	go test -tags bench -bench=./test/bench
```
Comment (hackerwins) [inline Makefile:42]:
```suggestion
	go test -tags bench -bench=. ./test/bench
```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution. It would be good to register additional issues to find bottlenecks on the bench.

I left a brief comment.

File: test/bench/editing_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	if err = json.Unmarshal(byteValue, &trace); err != nil {
+		b.Fatal(err)
+	}
+
+	return &trace, err
+}
+
+func BenchmarkEditing(b *testing.B) {
```
Comment (hackerwins) [inline test/bench/editing_bench_test.go:65]:
```suggestion
func BenchmarkTextEditing(b *testing.B) {
```

**Comment by zeikar** [comment]
@hackerwins 
Thank you for your reviews. I have committed all your suggestions.

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.

The test code doesn't seem to have any issues. I found an issue with a linear increase in execution time when running text edits.

```bash
    editing_bench_test.go:86: processing... 1000 833.432413ms
    editing_bench_test.go:86: processing... 2000 2.486363836s
    editing_bench_test.go:86: processing... 3000 3.766670666s
    editing_bench_test.go:86: processing... 4000 5.264808552s
    editing_bench_test.go:86: processing... 5000 7.45837556s
    ...
```

It might be a good idea to create a separate issue that finds the cause of this problem and fixes it.

**Pull request merged**